### PR TITLE
feat: jvm publishes to kafka and adds partition mapper

### PR DESCRIPTION
**Pull request by matt2e**
Removes jvm from pubsub integration tests as it is no longer consuming from where it is publishing.

### CODE REVIEW

File: examples/java/ExampleInterface.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+public interface ExampleInterface {
+    void instanceMethod();
```
Comment (stuartwdouglas) [inline examples/java/ExampleInterface.java:3]: What is this?

File: backend/controller/pubsub/integration_test.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ func TestPubSub(t *testing.T) {
 	calls := 20
 	events := calls * 10
 	in.Run(t,
-		in.WithLanguages("java", "go", "kotlin"),
```
Comment (stuartwdouglas) [inline backend/controller/pubsub/integration_test.go:22]: How far away is consumer support? Should we hold this until it is ready?

**Review by stuartwdouglas — APPROVED** [review]

File: examples/java/ExampleInterface.java (hunk 1 above)
Comment (matt2e) [reply examples/java/ExampleInterface.java:3]: 🤦‍♂️

File: backend/controller/pubsub/integration_test.go (hunk 2 above)
Comment (stuartwdouglas) [reply backend/controller/pubsub/integration_test.go:22]: What do you mean by:

> runner sends to kafka/redpanda, but also to the runner.

**Pull request merged**